pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -460,16 +462,17 @@ func clearExpectations(t *testing.T, manager *daemonSetsController, ds *apps.Dae
 }
 
 func TestDeleteFinalStateUnknown(t *testing.T) {
+	logger, ctx := ktesting.NewTestContext(t)
```

Comment:
I think it would be safer to not share the context between controller instances. Can you please go through the tests to find places like these and create a new context for each subtest? 

Or maybe better create the context in the `newTestController` function and have it as one of the return values?